### PR TITLE
add __getattribute__ to gf.Component

### PR DESCRIPTION
**Pull request by sebastian-goeldi**
This is without a doubt the grossest hack I probably ever did in a public repo.

Probably should use https://github.com/jayphelps/git-blame-someone-else

@joamatab 
@tvt173

### CODE REVIEW

**Comment by tvt173** [comment]
If we do something like this, wouldn't it be cleaner to just implement it as a property?

**Comment by sebastian-goeldi** [comment]
> If we do something like this, wouldn't it be cleaner to just implement it as a property?

If we were to do that, we would have to handle each property on it's own afaik. And the setter also as otherwise `c.xmin=5` will set it to 0.005um but getter will report `0.005`.

So I think for migration purposes the getattribute is cleaner.

**Comment by sebastian-goeldi** [comment]
Maybe properties on Instance would make more sense, but I am not sure. Also, this is really a dirty dirty hack. But this would work on Instances as well now.

File: gdsfactory/component.py [hunk 1]
```diff
@@ -113,6 +114,94 @@ def copy(self) -> kdb.Region:
         return self.dup()
 
 
+class Reference(kf.Instance):
```
Comment (joamatab) [inline gdsfactory/component.py:117]: can we call it ComponentReference?

Comment (sebastian-goeldi) [reply gdsfactory/component.py]: absolutely, I just named it something

Comment (joamatab) [reply gdsfactory/component.py]: awesome, i pushed

**Comment by sebastian-goeldi** [comment]
Added migration script used with

`gf migrate --migration 7to8 <input_folder/file> <output_folder/file>` to write to a new folder
`gf migrate -i --migration 7to8 <input_folder/file>` or `gf migrate --inplace --migration 7to8 <input_folder/file>` for in-place migration

**Comment by joamatab** [comment]
looks great! thank you Sebastian!

**Review by joamatab — APPROVED** [review]

**Pull request merged**